From 3da9428c67dd508d25b7beaf73c8ee61b65a483f Mon Sep 17 00:00:00 2001
From: Lionel Debieve <lionel.debieve@foss.st.com>
Date: Tue, 22 Mar 2022 13:38:48 +0100
Subject: [PATCH 0498/1695] drivers: firewall: add probe exception for LTDC

LTDC driver must be probed at startup even if the LTDC
is not secure. The LTDC is managed by non secure most of
the time but resources become secured when the TUI application
start.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git b02966e500d94e57f9eadc8478bacfa990b3f685 drivers: firewall: add probe exception for LTDC]
Change-Id: Ia6b3286d489bb717f9b91826935c0e38eb5330de
Signed-off-by: Lionel Debieve <lionel.debieve@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/245066
Reviewed-by: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/drivers/firewall/stm32_firewall.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/core/drivers/firewall/stm32_firewall.c b/core/drivers/firewall/stm32_firewall.c
index 630c3a59d..4b3c90f9f 100644
--- a/core/drivers/firewall/stm32_firewall.c
+++ b/core/drivers/firewall/stm32_firewall.c
@@ -19,7 +19,8 @@ static LIST_HEAD(fw_dev_list_head, stm32_firewall_device) fw_dev_list =
 static unsigned int list_lock = SPINLOCK_UNLOCK;
 
 static const char firewall_compatible_exceptions[][MAX_COMPAT_LENGTH] = {
-	"st,stm32mp15-i2c-non-secure"
+	"st,stm32mp15-i2c-non-secure",
+	"st,stm32-ltdc"
 };
 
 static struct stm32_firewall_device *find_device(paddr_t base, size_t size,
-- 
2.47.2

